From: Doug Ledford <dledford@redhat.com>
Date: Fri, 3 Sep 2010 21:20:18 -0400
Subject: [net] cxgb3: clear fatal parity error register on init
Message-id: <24577c391f2137bb5548bfbd2144feb1d954772d.1283548506.git.dledford@redhat.com>
Patchwork-id: 28139
O-Subject: [Patch RHEL5.6 2/4] cxgb3: now that we define fatal parity errors,
	make sure they are cleared
Bugzilla: 630124
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Related: bz630124

Signed-off-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/net/cxgb3/t3_hw.c b/drivers/net/cxgb3/t3_hw.c
index 0e2256a..b2147a6 100644
--- a/drivers/net/cxgb3/t3_hw.c
+++ b/drivers/net/cxgb3/t3_hw.c
@@ -3550,6 +3550,7 @@ int t3_init_hw(struct adapter *adapter, u32 fw_params)
 	t3_write_reg(adapter, A_PM1_TX_MODE, 0);
 	chan_init_hw(adapter, adapter->params.chan_map);
 	t3_sge_init(adapter, &adapter->params.sge);
+	t3_set_reg_field(adapter, A_PL_RST, 0, F_FATALPERREN);
 
 	t3_write_reg(adapter, A_T3DBG_GPIO_ACT_LOW, calc_gpio_intr(adapter));
 
